cluster-autoscaler/cloudprovider/rancher/rancher_node_group.go

```diff
@@ -0,0 +1,162 @@
+
+// NodeGroup contains configuration info and functions to control a set
+// of nodes that have the same capacity and set of labels.
+type rancherNodeGroup struct {
+	manager	*RancherManager
+	id		string
+}
+
+func (ng *rancherNodeGroup) nodePool() *v3.NodePool {
+	nodePool, err := ng.manager.GetCachedNodePool(ng.id)
+	if err != nil {
+		glog.Fatalf("failed to get node pool %s: %s", ng.id, err)
+	}
+
+	return nodePool
+}
+
+// MaxSize returns maximum size of the node group.
+func (ng *rancherNodeGroup) MaxSize() int {
+	nodePool := ng.nodePool()
+
+	if maxSizeValue, exists := nodePool.Annotations[maxSizeAnnotation]; exists {
+		maxSize, err := strconv.Atoi(maxSizeValue)
+		if err != nil {
+			glog.Errorf("failed to parse max size from node pool %s: %s", nodePool.HostnamePrefix, err)
+			return defaultMaxSize
+		}
+
+		return maxSize
+	}
+
+	return defaultMaxSize
+}
+
+// MinSize returns minimum size of the node group.
+func (ng *rancherNodeGroup) MinSize() int {
+	nodePool := ng.nodePool()
+
+	if minSizeValue, exists := nodePool.Annotations[minSizeAnnotation]; exists {
+		minSize, err := strconv.Atoi(minSizeValue)
+		if err != nil {
+			glog.Errorf("failed to parse min size from node pool %s: %s", nodePool.HostnamePrefix, err)
+			return defaultMinSize
+		}
+
+		return minSize
+	}
+
+	return defaultMinSize
+}
+
+// TargetSize returns the current target size of the node group. It is possible that the
+// number of nodes in Kubernetes is different at the moment but should be equal
+// to Size() once everything stabilizes (new nodes finish startup and registration or
+// removed nodes are deleted completely). Implementation required.
+func (ng *rancherNodeGroup) TargetSize() (int, error) {
+	return int(ng.nodePool().Quantity), nil
+}
+
+// IncreaseSize increases the size of the node group. To delete a node you need
+// to explicitly name it and use DeleteNode. This function should wait until
+// node group size is updated. Implementation required.
+func (ng *rancherNodeGroup) IncreaseSize(delta int) error {
+	return ng.manager.IncreasePoolSize(ng.id, delta)
+}
+
+// DeleteNodes deletes nodes from this node group. Error is returned either on
+// failure or if the given node doesn't belong to this node group. This function
+// should wait until node group size is updated. Implementation required.
+func (ng *rancherNodeGroup) DeleteNodes(nodes []*v1.Node) error {
+	return ng.manager.DeleteNodes(ng.id, nodes)
+}
+
+// DecreaseTargetSize decreases the target size of the node group. This function
+// doesn't permit to delete any existing node and can be used only to reduce the
+// request for new nodes that have not been yet fulfilled. Delta should be negative.
+// It is assumed that cloud provider will not delete the existing nodes when there
+// is an option to just decrease the target. Implementation required.
+func (ng *rancherNodeGroup) DecreaseTargetSize(delta int) error {
+	return fmt.Errorf("sry but I won't decrease the target size ¯\\_(ツ)_/¯")
```

Comment:
Supposing that cluster-autoscaler needed to invoke this method, will it delete the extra nodes after its creation?

btw there's an extra `\` character in your sassy emoji. ahahah